staker/staker.go

```diff
@@ -383,10 +373,26 @@ func (s *Staker) tryFastConfirmationNodeNumber(ctx context.Context, number uint6
 }
 
 func (s *Staker) tryFastConfirmation(ctx context.Context, blockHash common.Hash, sendRoot common.Hash) error {
-	if !s.config.EnableFastConfirmation {
+	if s.fastConfirmer == (common.Address{}) {
 		return nil
 	}
-	if s.fastConfirmSafe != nil {
+	// Only use gnosis safe fast confirmation, if the safe address is different from the wallet address, else it's not a safe contract.
+	if s.fastConfirmer != *s.wallet.Address() {
+		// If we don't have a fastConfirmSafe, create one
+		if s.fastConfirmSafe == nil {
+			fastConfirmSafe, err := NewFastConfirmSafe(
```

Comment:
Let's do this in the constructor. Also let's make sure that `s.wallet.Address()` isn't nil.

Reply:
Moved to constructor. Also now using AddressOrZero(), should have the same effect.